### PR TITLE
[AIRFLOW-3948] GCSToBQOperator list as templating field issue

### PR DESCRIPTION
**Pull request by ryanyuan**
Fix GCSToBQOp list templating
Add unit tests for GCSToBQOp

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow-3948](https://issues.apache.org/jira/browse/AIRFLOW-3948) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
tests/contrib/operators/test_gcs_to_bq_operator.py

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/operators/gcs_to_bq.py [hunk 1]
```diff
@@ -24,6 +24,11 @@
 from airflow.models import BaseOperator
 from airflow.utils.decorators import apply_defaults
 
+try:
+    basestring
```
Comment (mik-laj) [inline airflow/contrib/operators/gcs_to_bq.py]: Is there any reason why you have not used the `six` package here? Using this package makes it easier to track where we have the code for the older version of Python. We plan to drop support for the older version, so we will simply delete all references to the `six` package.

Comment (ryanyuan) [reply airflow/contrib/operators/gcs_to_bq.py]: @mik-laj Thanks for the suggestion. I've updated my code.

**Comment by Fokko** [comment]
Wouldn't this be an issue for templated lists in general?

**Comment by ryanyuan** [comment]
> Wouldn't this be an issue for templated lists in general?

@Fokko That's true. My suggestion is to move the conversion function to utils.helpers and perform conversion for those jinja templating fields that are of list type. What do you think?

**Comment by Fokko** [comment]
I would prefer to solve this on a global level, yes! 👍

**Pull request closed without merging**